packages/blade/src/components/Input/BaseInput/BaseInputVisuals.tsx

```diff
@@ -12,11 +12,12 @@ type InputVisuals = Pick<
 
 const getVisualContainerStyles = (): Pick<
   BaseBoxProps,
-  'display' | 'flexDirection' | 'alignItems'
+  'display' | 'flexDirection' | 'alignItems' | 'alignSelf'
 > => ({
   display: 'flex',
   flexDirection: 'row',
   alignItems: 'center',
+  alignSelf: 'center',
```

Comment:
aligns the trailing icons to center like before. The trailing container's size was made full height to make chevron icon clickable from anywhere. So the elements inside of it got top-aligned. By setting alignSelf: center here, we make them center again while keeping height full.
 
<img width="304" alt="image" src="https://github.com/razorpay/blade/assets/30949385/16cda5d1-12f0-4e06-a181-97b614bdf45c">